doc/changelog.rst

```diff
@@ -0,0 +1,84 @@
+Changelog
+=========
+
+1.0.0
+-----
+
+Warning: the 1.0.0 is a breaking release from the beta 0.X.X series.
+
+As compared to the prior series (0.X.X), some names and module organizations have changed. This
+means existing code that uses Bycycle may no longer work as currently written with the new version,
+and may need updating. You should update to the new version when you are ready to update your code
+to reflect the new changes.
+
+Note that the main changes are in code organization, some names, and the addition of many new
+features. The algorithm itself has not changed, and results from the new version should be roughly
+equivalent to those with older versions, with a few minor exceptions.
+
+Code Oraganization
+~~~~~~~~~~~~~~~~~~
+
+- Increased modularity.
+
+  - burst: ``burst.amp``, ``burst.cycle``, ``burst.dualthresh``, ``burst.utils``
+  - cyclepoints: ``cyclepoints.extrema``, ``cyclepoints.zerox``, ``cyclepoints.phase``
+  - features: ``features.features``, ``features.burst``, ``features.shape``, ``features.cyclepoints``
+  - The dataframe output returned from ``compute_features`` now returns:
+
+    1. ``df_features``: burst/shape cycle features.
+    2. ``df_samples``: cyclepoint locations as signal indices.
+
+- Neurodsp dependency.
+
+  - `filt.py` and `sim.py` have been replaced with equivalent neurodsp functions.
+
+Naming Updates
+~~~~~~~~~~~~~~
+
+- Consistency with neurodsp and PEP8 compliance:
```

Comment:
I actually don't think it's all that important to split this up by different reasons. From the users perspective, what matters is these have all changed (there is nothing actionable different about the different groups). I would probably just say:
`The following argument names have been changed throughout the module:`
(then one big list)